modules/schema/src/main/java/org/apache/ignite/internal/schema/BinaryTupleParser.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.schema;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import org.apache.ignite.lang.IgniteInternalException;
+
+/**
+ * Binary tuple parser allows to get bytes of individual elements from entirety of tuple bytes.
+ */
+public class BinaryTupleParser {
+    /**
+     * Receiver of parsed data.
+     */
+    public interface Sink {
+        /**
+         * Provides the location of the next tuple value.
+         *
+         * @param index Value index.
+         * @param begin Start offset of the value, 0 for NULL.
+         * @param end End offset of the value, 0 for NULL.
+         */
+        void nextElement(int index, int begin, int end);
```

Comment:
It looks like this method is used for instance initialization.
The index param is never used. Can Sink have a package visibility?